plugins/woocommerce/src/Internal/DataStores/Orders/OrdersTableMetaQuery.php

```diff
@@ -148,31 +148,32 @@ private function is_atomic( array $arg ): bool {
 	private function sanitize_meta_query( array $q ): array {
 		$sanitized = array();
 
-		if ( $this->is_atomic( $q ) ) {
-			if ( isset( $q['value'] ) && array() === $q['value'] ) {
-				unset( $q['value'] );
-			}
-
-			$q['compare']     = isset( $q['compare'] ) ? strtoupper( $q['compare'] ) : ( isset( $q['value'] ) && is_array( $q['value'] ) ? 'IN' : '=' );
-			$q['compare_key'] = isset( $q['compare_key'] ) ? strtoupper( $q['compare_key'] ) : ( isset( $q['key'] ) && is_array( $q['key'] ) ? 'IN' : '=' );
-
-			if ( ! in_array( $q['compare'], self::NON_NUMERIC_OPERATORS, true ) && ! in_array( $q['compare'], self::NUMERIC_OPERATORS, true ) ) {
-				$q['compare'] = '=';
-			}
-
-			if ( ! in_array( $q['compare_key'], self::NON_NUMERIC_OPERATORS, true ) ) {
-				$q['compare_key'] = '=';
-			}
-
-			return $q;
+		if ( ! is_array( $q ) ) {
```

Comment:
Wondering, do we need this check? The method signature already specifies that `$q` should be an array so I'm guessing if a non-array is passed it will either be coerced into an array or an error will be raised and execution won't reach this point. wdyt?

Reply:
Yes, you're right. I'm still old school in this regard and always forget about type hints 😅